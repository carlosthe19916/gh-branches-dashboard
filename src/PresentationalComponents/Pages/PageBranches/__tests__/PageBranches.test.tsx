import React from "react";
import { shallow } from "enzyme";
import queryString from "query-string";
import {
  PageBranches,
  PageBranchesProps,
  extractBranchOrderQueryParam
} from "../PageBranches";

it("Test extractBranchOrderQueryParam :: no param", () => {
  const queryParams = queryString.parse("");
  const result = extractBranchOrderQueryParam(queryParams);

  expect(result).toMatchObject([]);
});

it("Test extractBranchOrderQueryParam :: single param", () => {
  const queryParams = queryString.parse("?branchOrder=branch1");
  const result = extractBranchOrderQueryParam(queryParams);

  expect(result).toMatchObject(["branch1"]);
});

it("Test extractBranchOrderQueryParam :: multiple params", () => {
  const queryParams = queryString.parse(
    "?branchOrder=branch1&branchOrder=branch2"
  );
  const result = extractBranchOrderQueryParam(queryParams);

  expect(result).toMatchObject(["branch1", "branch2"]);
});

it("Test snapshot", () => {
  const props: PageBranchesProps = {
    match: {
      params: {
        owner: "carlosthe19916",
        repository: "gh-branches-dashboard"
      }
    },
    history: jest.fn(),
    location: {
      seach: "?"
    }
  };

  const wrapper = shallow(<PageBranches {...props} />);
  expect(wrapper).toMatchSnapshot();
});
