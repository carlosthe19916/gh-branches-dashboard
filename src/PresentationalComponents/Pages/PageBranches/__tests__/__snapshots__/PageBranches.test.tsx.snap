// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<Fragment>
  <Connect(ContextRepositoryLoader)
    repositoryId="carlosthe19916/gh-branches-dashboard"
  >
    <PageSection
      variant="light"
    >
      <TextContent>
        <Text
          component="h1"
        >
          Branches
        </Text>
        <Text
          component="p"
        >
          Monitor your branches here.
        </Text>
      </TextContent>
    </PageSection>
    <PageSection>
      <Bullseye>
        <Connect(BranchesBoard)
          userDefinedBranchOrder={Array []}
        />
      </Bullseye>
    </PageSection>
  </Connect(ContextRepositoryLoader)>
</Fragment>
`;
